packages/web-app-files/src/helpers/resources.js

```diff
@@ -63,8 +63,8 @@ export function buildResource(resource) {
       return this.permissions.indexOf('C') >= 0
     },
     canDownload: function() {
-      // TODO: as soon as we allow folder downloads as archive we want to return `true` here without exceptions
-      return !isFolder
+      // TODO: as soon as we allow folder downloads as archive we want to remove !isFolder from this test
+      return !isFolder && this.permissions.indexOf('V') >= 0
```

Comment:
Looking at https://github.com/owncloud/core/blob/d41294c2d1d97138e1ad4f978e7f01dc83abc73f/apps/dav/lib/Connector/Sabre/Node.php#L338 I think `V` is the permission for `moveable`. 🤔 Where did you get the download info from? It's poorly documented... I can't a complete list anywhere.

Reply:
Yep it's very hard to find, it was added via the interface and WebDAV as a PR, it works fine but not well documented.  It's a custom permission supported by `owncloud/core`.  It reads it from the attributes on the share as follows:
`attributes[{ scope: 'permissions', key: 'download', enabled: true }]`

V is moveable and appears to be the only permission which returns false once download is enabled, so this is the only option in terms of these list of permissions to choose from.  It would be better if there were a an 'O' permission but that's outside of this scope.

In WebDAV clients, the file can be _seen_ (ie. when listing the folder contents) but not downloaded.

Open to any better suggestion but I could not find another way to implement this.

Reply:
Good that you mention the attributes, learned about them just a few days ago. 😁 The attributes are available via SDK as far as I know, so we could implement this via attributes. If I understand you correctly, `download` attribute being enabled means that all other permissions get disabled? 🤔

Reply:
I don't believe so, you can find references in the codebase (and OO integration) if it is defined (and is set to false) only then is download prevented:

https://github.com/owncloud/core/pull/33994/files#diff-00138679c99de6bf1310d87711a09ae77a1f0ad5cad42fee0682f7b28c9de79aR105-R109
https://github.com/ONLYOFFICE/onlyoffice-owncloud/blob/master/controller/editorcontroller.php#L896-L899

Otherwise the typical read permission is used, as has been implemented in the **owncloud/core** historically.

Reply:
If there is a way to access these extended attributes without an API call here, let me know and i'll adjust the PR.

Reply:
@kulmann i'm just reviewing my PRs in this repo, where did we land with the thinking on this one?

Reply:
Thanks for coming back to this! The attributes already get fetched, so in as far as I can tell you could modify https://github.com/owncloud/web/blob/4d58f7a6e2112d2af6d8e8101895d461599ccbf7/packages/web-app-files/src/helpers/resources.js#L297 to check if `resource.attributes.find(a => a.scope === 'permissions'  && a.key === 'dwnload')` is not nullish and has `enabled` set to `false`. What do you think? 🤔  Can take this over if you want to.

Reply:
That sounds good, I haven't kept up to date with changes in the codebase (the original PR checked the 'V' permission) but should be easy enough to test, and as long as it works with an OC10 backend it should be fine.

Yes, if you can take this over that would be appreciated.